src/Stratis.Bitcoin.Features.PoA/Voting/JoinFederationRequestBuilder.cs

```diff
@@ -50,6 +49,9 @@ public static JoinFederationRequest Deconstruct(Transaction trx, JoinFederationR
             if (ops[0].Code != OpcodeType.OP_RETURN)
                 return null;
 
+            if (trx.Outputs[1].Value.ToDecimal(MoneyUnit.BTC) != VotingRequestTransferAmount)
```

Comment:
Does returning null here inform the user of the invalid request amount?

Reply:
This is not a value that is supplied by the user and is regarded as part of the format of the voting request. A consensus error is raised in the `VotingRequestValidationRule` if transaction in any way diverts from the expected format. The only way to provide a different amount is when a hacker builds his/her own custom voting request. As such we don't care to return any specific details to the hacker.

Reply:
OK yes makes sense!